operate/qa/integration-tests/src/test/java/io/camunda/operate/elasticsearch/FinishedImportingIT.java

```diff
@@ -182,6 +172,47 @@ final var record = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 1);
         .until(() -> isRecordReaderIsCompleted("2-process-instance"));
   }
 
+  @Test
+  public void shouldNotSetCompletedToFalseForSubsequentRecordsAfterImportingDone()
+      throws IOException {
+    final var record = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 1);
+    EXPORTER.export(record);
+    esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+
+    zeebeImporter.performOneRoundOfImport();
+
+    // when
+    final var record2 = generateRecord(ValueType.PROCESS_INSTANCE, "8.7.0", 1);
+    EXPORTER.export(record2);
+    esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+
+    // receives 8.7 record and marks partition as finished importing
+    zeebeImporter.performOneRoundOfImport();
+
+    // then
+    // require multiple checks to avoid race condition. If records are written to zeebe indices and
+    // before a refresh, the record reader pulls the import batch is empty so it then says that the
+    // record reader is done when it is not.
```

Comment:
🔧 
```suggestion
    // Require multiple checks to avoid race condition. 
    // Otherwise: If records are written to zeebe indices and before a refresh, the record reader pulls an empty import batch, then it might assume falsely
    // that it is done, while it is not.
```